api/src/main/java/org/apache/cloudstack/api/command/user/template/CreateTemplateCmd.java

```diff
@@ -139,6 +138,15 @@ public class CreateTemplateCmd extends BaseAsyncCreateCmd implements UserCmd {
     @Parameter(name = ApiConstants.ZONE_ID, type = CommandType.UUID, entityType = ZoneResponse.class, description = "the zone for the template. Can be specified with snapshot only", since = "4.19.0")
     private Long zoneId;
 
+    @Parameter(name = ApiConstants.DOMAIN_ID,
+          type = CommandType.UUID,
+          entityType = DomainResponse.class,
+          description = "an optional domainId. If the account parameter is used, domainId must also be used.")
```

Comment:
maybe add a `since` property here

Reply:
Thanks @shwstppr for reviewing this PR. It's included in latest commit.